cmd/installer/cli/install.go

```diff
@@ -839,15 +839,50 @@ func installAndEnableLocalArtifactMirror() error {
 	if _, err := helpers.RunCommand("systemctl", "daemon-reload"); err != nil {
 		return fmt.Errorf("unable to get reload systemctl daemon: %w", err)
 	}
+	if _, err := helpers.RunCommand("systemctl", "enable", "local-artifact-mirror"); err != nil {
+		return fmt.Errorf("unable to enable the local artifact mirror service: %w", err)
+	}
+	logrus.Debugf("Starting local artifact mirror")
 	if _, err := helpers.RunCommand("systemctl", "start", "local-artifact-mirror"); err != nil {
 		return fmt.Errorf("unable to start the local artifact mirror: %w", err)
 	}
-	if _, err := helpers.RunCommand("systemctl", "enable", "local-artifact-mirror"); err != nil {
-		return fmt.Errorf("unable to start the local artifact mirror service: %w", err)
+	if err := waitForLocalArtifactMirror(ctx); err != nil {
+		return fmt.Errorf("unable to wait for the local artifact mirror: %w", err)
 	}
+	logrus.Debugf("Local artifact mirror started!")
 	return nil
 }
 
+func waitForLocalArtifactMirror(ctx context.Context) error {
+	consecutiveSuccesses := 0
+	requiredSuccesses := 3
+	maxAttempts := 30
+	checkInterval := 2 * time.Second
+
+	var lastErr error
+	for attempt := 0; attempt < maxAttempts; attempt++ {
+		_, err := helpers.RunCommand("systemctl", "status", "local-artifact-mirror")
```

Comment:
should this use the same expbackoff with jitter as we do elsewhere? I see we're looking for consecutive successes and I'm not sure if that would work though